File: Base/src/main/java/com/hx/steven/app/MyApp.java
package com.hx.steven.app;

import com.alibaba.android.arouter.launcher.ARouter;
import com.hx.steven.BuildConfig;

/**
 * Created by user on 2018/1/15.
 */

public class MyApp extends BaseApplication {

    @Override
    public AppInitBuilder getAppInitBuilder() {
        return new AppInitBuilder.Builder()
                .setInitLogger(true)
                .build();
    }

    @Override
    public void onCreate() {
        super.onCreate();
//        try {
//            RetrofitNetManager.getInstance().init("https://www.wanandroid.com");
//            RetrofitNetManager.getInstance().setApiService(RetrofitNetManager.getInstance().getRetrofit().create(ApiService.class));
//        } catch (Exception e) {
//            e.printStackTrace();
//        }
        if (BuildConfig.DEBUG) {
            ARouter.openLog();
            ARouter.openDebug();
        }
        ARouter.init(this);
    }

    @Override
    public void onTerminate() {
        super.onTerminate();
        ARouter.getInstance().destroy();
    }
}

File: Base/src/main/java/com/hx/steven/app/MyApp.kt
package com.hx.steven.app

import com.hx.steven.di.allModule
import com.hx.steven.util.CrashHandlerUtil
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin

/**
 * Created by user on 2018/1/15.
 */
class MyApp : BaseApplication() {

    override val appInitBuilder: AppInitBuilder
        get() = AppInitBuilder.Builder()
            .setInitLogger(true)
            .setOkGo(true)
            .setMMKV(true)
            .setAudio(true)
            .build()

    override fun onCreate() {
        // 初始化koin
        startKoin {
            androidLogger()
            androidContext(this@MyApp)
            modules(allModule)
        }

        val mException: CrashHandlerUtil = CrashHandlerUtil.getInstance()
        mException.init(this)
        super.onCreate()
    }
}